Neos.ContentRepositoryRegistry/Classes/ContentRepositoryRegistry.php

```diff
@@ -81,6 +81,14 @@ public function get(ContentRepositoryId $contentRepositoryId): ContentRepository
         return $this->getFactory($contentRepositoryId)->getOrBuild();
     }
 
+    /**
+     * @return array<ContentRepositoryId>
+     */
+    public function getContentRepositoryIds(): array
+    {
+        return array_map(ContentRepositoryId::fromString(...), array_keys($this->settings['contentRepositories']));
+    }
```

Comment:
```suggestion
    /**
     * @return iterable<ContentRepositoryId>
     */
    public function getContentRepositoryIds(): iterable
    {
        return array_map(ContentRepositoryId::fromString(...), array_keys($this->settings['contentRepositories'] ?? []));
    }
```

this way we can easily introduce a collection dto at some point and it wont crash if nothing is configured.

Reply:
Resolved with #5194